force-app/main/default/classes/RD2_RecurringDonation.cls

```diff
@@ -306,6 +316,22 @@ public inherited sharing class RD2_RecurringDonation {
         }
     }
 
+    public RD2_RecurringDonation statusAutomationReviseStatus(RD2_ScheduleService scheduleService) {
+        RD2_StatusAutomationService statusAutomationServce = new RD2_StatusAutomationService();
+
+        if(statusAutomationServce.shouldMarkRDAsLapsed(this, scheduleService)) {           
```

Comment:
Could we check `shouldMarkRDAsClosed` first, and then turn this into an `if/else if` statement?